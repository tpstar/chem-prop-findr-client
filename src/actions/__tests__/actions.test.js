import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { fetchChemical } from '../index';

const middlewares = [ thunk ];
const mockStore = configureMockStore(middlewares)

const store = mockStore();

describe('fetchChemical', () => {

  beforeEach(() => {
    store.clearActions();
  });
  
  it('should handle FETCH_CHEMICAL_SUCCESS action', async () => {
    const response = '{"id": 1, "name": "Fake Toluene", "fw": "92.14", "formula": "C6H5CH3"}';
    fetch.mockResponseSuccess(response);
    await store.dispatch(fetchChemical('toluene'));
    //store.getActions gives an array of actions
    expect(store.getActions()).toMatchSnapshot()
  })
})
