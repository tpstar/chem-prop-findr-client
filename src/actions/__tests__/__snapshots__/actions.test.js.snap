// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchChemical should handle FETCH_CHEMICAL_SUCCESS action 1`] = `
Array [
  Object {
    "type": "FETCH_CHEMICAL_REQUEST",
  },
  Object {
    "payload": Object {
      "formula": "C6H5CH3",
      "fw": "92.14",
      "id": 1,
      "name": "Fake Toluene",
    },
    "type": "FETCH_CHEMICAL_SUCCESS",
  },
]
`;
